### PR TITLE
add an animation for scrollTop

### PR DESCRIPTION
**Pull request by jDramaix**

### CODE REVIEW

File: src/main/java/com/arcbees/geekette/client/application/ApplicationView.java [hunk 1]
```diff
@@ -28,6 +33,34 @@
...
+        @Override
+        protected void onStart() {
+            start = element.getScrollTop();
+            super.onStart();
+        }
+
+        @Override
+        protected void onUpdate(double progress) {
+            double value = start - start * progress;
+            element.setScrollTop((int) value);
```
Comment (jDramaix) [inline src/main/java/com/arcbees/geekette/client/application/ApplicationView.java:53]: Normally this animation should be implemented in the presenter side and the presenter should call the view only to set the scrollTop... Feel free to fix that if you want

File: src/main/java/com/arcbees/geekette/client/application/ApplicationView.java [hunk 2]
```diff
@@ -38,4 +71,15 @@
     void onBuy(ClickEvent event) {
         getUiHandlers().onBuy();
     }
+
+    @UiHandler("backToTop")
+    void onBackToTopClicked(ClickEvent e) {
+        GQuery.$("html, body").each(new Function() {
```
Comment (jDramaix) [inline src/main/java/com/arcbees/geekette/client/application/ApplicationView.java:77]: GQuery should be able to automatically animate the scrollTop by `$("html, body").animate("'$scrollTop':0")`But it seems that the last changes that Manolo did to implement css3 animation breaks that.

Comment (christiangoudreau) [inline src/main/java/com/arcbees/geekette/client/application/ApplicationView.java:77]: Let's fix it then :P

**Comment by christiangoudreau** [comment]
LGTM, that will be enough for now

**Pull request merged**